src/DynamoPackages/PackageLoader.cs

```diff
@@ -102,6 +103,36 @@ public PackageLoader(IEnumerable<string> packagesDirectories)
             packagesDirectoriesToVerifyCertificates.Add(StandardLibraryDirectory);
         }
 
+        internal void OnWorkspaceRead(IEnumerable<INodeLibraryDependencyInfo> packageDependencies, 
+            IEnumerable<string> packageDirectories)
+        {
+            var pkgs = new List<Package>();
+            foreach (var root in packageDirectories)
+            {
+                foreach (var dir in
+                    Directory.EnumerateDirectories(root, "*", SearchOption.TopDirectoryOnly))
+                {
+                    var headerPath = Path.Combine(dir, "pkg.json");
+                    if (PathHelper.IsValidPath(headerPath))
+                    {
+                        var discoveredPkg = Package.FromJson(headerPath, AsLogger());
+                        if(discoveredPkg == null) continue;
+
+                        foreach (var info in packageDependencies)
+                        {
+                            // if workspace package dependency is present in downloaded packages, load it.
+                            if (info.Name == discoveredPkg.Name && info.Version.ToString() == discoveredPkg.VersionName)
```

Comment:
Collect all downloaded packages that match with the ones referenced in the workspace.